guava/src/com/google/common/math/Quantiles.java

```diff
@@ -513,7 +543,12 @@ private static void checkIndex(int index, int scale) {
    * ({@code required}, {@code to}] are greater than or equal to that value. Therefore, the value at
    * {@code required} is the value which would appear at that index in the sorted dataset.
    */
-  private static void selectInPlace(int required, double[] array, int from, int to) {
+  @SuppressWarnings(value = {"lowerbound:assignment.type.incompatible",/*(1): When entering the loop,
+          required > from and from >= 0, therefore required >= 1.
+          At the assignment, partitionPoint >= required, therefore partitionPoint - 1 >= 0. */
+          "upperbound:assignment.type.incompatible"/*(2): Entering the loop, from and to < array.length. While to > from,
+          to > partitionPoint + 1, therefore partitionPoint + 1 < array.length */})
```

Comment:
It seems to me that after `int partitionPoint = partition(array, from, to);`, it can be that `partitionPoint == to`, contrary to this comment. Is that right?
Maybe you mean that it holds at certain point in the program? However, that is not stated in the comment, and it is not obvious. For someone trying to understand the code, it would be just confusing.
I definitely want to avoid statements that are, or look like, a contradiction.
If you know why the code is correct, can you write explicitly all the steps you used to assert that? It may be longer, but better a longer clear explanation than a short confusing one.

Reply:
Please let me know if the new replaced comment makes better sense.